nautobot/docs/plugins/development.md

```diff
@@ -608,6 +610,90 @@ secrets_providers = [ConstantValueSecretsProvider]
+
+The dynamically-added lookup expressions can be found in the source code at [nautobot/utilities/constants.py](https://github.com/nautobot/nautobot/blob/main/nautobot/utilities/constants.py) and the mapping logic can be found in [nautobot/utilities/filters.py](https://github.com/nautobot/nautobot/blob/main/nautobot/utilities/filters.py). Please see the documentation on [filtering](../../rest-api/filtering/#lookup-expressions) for more information.
+
+!!! tip
+    For developers of plugins that define their own model filters, note that the above are added dynamically, as long as the class inherits from `nautobot.utilities.filters.BaseFilterSet`.
+
+However, that does not cover every possible use case, to list a few examples:
+
+* Usage of a custom `method` argument on a filter that points to a `FilterSet` method, which would allow arbitrary filtering using custom logic. This is how the `q` field search logic is currently performed.
+* Creation of a filter on a field that does not currently have filtering support
+* Convenience methods for highly nested fields
+
+There are several conditions that must be met in order to extend a filter:
+
+* The original FilterSet must follow the pattern: `f"{model.__name__}FilterSet"` e.g. `TenantFilterSet`
+* The `PluginFilterExtension.filterset_fields` attribute must be a valid dict, with each key being the filter name (which must start with the plugin's `name` + `_`, e.g. `"example_plugin_description"`, not merely `"description"`) and each value being a valid [django-filter](https://django-filter.readthedocs.io/en/main/) filter
+
+Nautobot will dynamically generate the additional relevant lookup expressions of a plugin's defined custom FilterSet field, so no need to additionally register `example_plugin_description__ic`, etc.
+
+Similar to `FilterSet` fields, Nautobot provides a default filter form for each model, however that does not cover every possible use case. To list a few examples of why one may want to extend a filter form:
+
+* The base filter form does not include a custom filter defined by the plugin as described above
+* The base filter form does not provide a specific lookup expression to a filterable field, such as allowing regex on name
+
+There are several conditions that must be met in order to extend a filter:
+
+* The original `FilterForm` must follow the pattern: `f"{model.__name__}FilterForm"`, e.g. `TenantFilterForm`
+* The `filterform_fields` attribute must be a valid dictionary of Django form fields
+
+!!! note
+    A plugin is not required to define both `filterset_fields` and `filterform_fields`.
+
+Example:
+
+```python
```

Comment:
TODO: Change to link to repo code example now that it's in a more permanent branch.